pyogrio/geopandas.py

```diff
@@ -81,14 +81,14 @@ def read_dataframe(
     except ImportError:
         raise ImportError("geopandas is required to use pyogrio.read_dataframe()")
 
-    path = str(path)
-
-    if not "://" in path:
-        if not "/vsi" in path.lower() and not os.path.exists(path):
-            raise ValueError(f"'{path}' does not exist")
+    if isinstance(path_or_buffer, str):
```

Comment:
We will want a similar check for `pathlib.Path` objects too (which we achieved previously by forcing everything to string).

Reply:
This check about the path existing (if a local file), is there a reason this lives specifically in `geopandas.py`, or we could also move it to `read_raw`, and here just pass through the `path_or_buffer`?

Reply:
Yes, this could move to `read_raw`; no reason it specifically lives here.